### PR TITLE
Implemented random functionality on button pressed

### PR DESCRIPTION
**Pull request by FranciscoGaraventa**
rollDiceLoader

### CODE REVIEW

File: lib/src/bloc/dice_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  final int maxNumDice = 6;
+  int _diceRoll = 0;
+  final _diceStateController = StreamController<int>();
+
+  StreamSink<int> get _inDiceRoll => _diceStateController.sink;
+
+  Stream<int> get diceRoll => _diceStateController.stream;
+  final _diceEventController = StreamController<DiceEvent>();
+
+  Sink<DiceEvent> get diceEventSink => _diceEventController.sink;
```
Comment (luisjavierglb) [inline lib/src/bloc/dice_bloc.dart]: I wouldnt provide a way from clients of this bloc to add items to the sink, we should eb exposing a mothod just to trigger the event we want, in that way we can control the input, makes sense ?

**Review by luisjavierglb — CHANGES_REQUESTED** [review]

File: lib/src/styles/dimensions.dart [hunk 2]
```diff
@@ -0,0 +1,4 @@
+abstract class Dimension{
+  static const double diceSizeBoxHeight = 20;
+  static const double rollWidgetTextFontSize = 30;
+}
```
Comment (MarcosCGlobant) [inline lib/src/styles/dimensions.dart]: Please add an end line space

File: lib/src/screen/dice/roll_widget.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            case ConnectionState.waiting:
+              return CircularProgressIndicator();
+            case ConnectionState.done:
+              if (snapshot.hasError)
+                return Text('Error: ${snapshot.error}');
+              else if (snapshot.hasData) {
+                return Text(
+                  '${snapshot.data}',
+                  style: TextStyle(
+                      fontWeight: FontWeight.bold, fontSize: Dimension.rollWidgetTextFontSize),
```
Comment (MarcosCGlobant) [inline lib/src/screen/dice/roll_widget.dart]: check for trailing commas, font size and font weight should be on different lines

File: lib/src/screen/dice/dice_screen.dart [hunk 4]
```diff
@@ -1,17 +1,31 @@
+import 'dart:async';
+import 'package:first_flutter_app/src/styles/dimensions.dart';
```
Comment (MarcosCGlobant) [inline lib/src/screen/dice/dice_screen.dart]: use relative imports for your own classes/files

File: lib/src/bloc/dice_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+import 'dart:math';
+import 'package:flutter/material.dart';
+
+import '../screen/dice/dice_event.dart';
+import 'package:first_flutter_app/src/bloc/bloc.dart';
```
Comment (MarcosCGlobant) [inline lib/src/bloc/dice_bloc.dart]: Use relative imports for your own classes/files

**Review by MarcosCGlobant — DISMISSED** [review]

File: lib/src/bloc/dice_bloc.dart (hunk 1 above)
Comment (FranciscoGaraventa) [reply lib/src/bloc/dice_bloc.dart]: Done.

File: lib/src/styles/dimensions.dart (hunk 2 above)
Comment (FranciscoGaraventa) [reply lib/src/styles/dimensions.dart]: Done!

File: lib/src/screen/dice/roll_widget.dart (hunk 3 above)
Comment (FranciscoGaraventa) [reply lib/src/screen/dice/roll_widget.dart]: Done!

File: lib/src/screen/dice/dice_screen.dart (hunk 4 above)
Comment (FranciscoGaraventa) [reply lib/src/screen/dice/dice_screen.dart]: Done!

File: lib/src/bloc/dice_bloc.dart (hunk 5 above)
Comment (FranciscoGaraventa) [reply lib/src/bloc/dice_bloc.dart]: Done!

File: lib/src/styles/dimensions.dart [hunk 6]
```diff
@@ -0,0 +1,6 @@
+abstract class Dimension {
+  static const double diceSizeBoxHeight = 20;
+  static const double rollWidgetTextFontSize = 30;
+  static const double diceBorderRadius = 20;
+
```
Comment (MarcosCGlobant) [inline lib/src/styles/dimensions.dart]: remove extra line

File: lib/src/bloc/dice_bloc.dart (hunk 5 above)
Comment (MarcosCGlobant) [reply lib/src/bloc/dice_bloc.dart:4]: this one still needs to be fixed

File: lib/src/bloc/dice_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  int _diceRoll = 0;
+  final _diceStateController = StreamController<int>();
+
+  StreamSink<int> get _inDiceRoll => _diceStateController.sink;
+
+  Stream<int> get diceRoll => _diceStateController.stream;
+  final _diceEventController = StreamController<DiceEvent>();
+
+  DiceBloc() {
+    _diceEventController.stream.listen(_mapEventToState);
```
Comment (luisjavierglb) [inline lib/src/bloc/dice_bloc.dart]: This does not add any value, we can remove that

File: lib/src/screen/dice/dice_screen.dart [hunk 8]
```diff
@@ -1,17 +1,33 @@
...
+  })  : assert(diceBloc != null),
+        super(key: key);
   final String title;
+  final DiceBloc diceBloc;
 
   @override
   _DiceScreenState createState() => _DiceScreenState();
 }
 
+Future<int> fetchValue(int roll) async {
```
Comment (luisjavierglb) [inline lib/src/screen/dice/dice_screen.dart]: This could be removed as well

File: lib/src/bloc/dice_bloc.dart [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        break;
+    }
+    _inDiceRoll.add(_diceRoll);
+  }
+
+  @override
+  void initialize() {}
+
+  void newEvent(DiceEvent event) {
+    _diceEventController.sink.add(event);
```
Comment (luisjavierglb) [inline lib/src/bloc/dice_bloc.dart]:
```
Future.delayed(const Duration(milliseconds: 500), () {
    _diceEventController.sink.add(event);
});
```

File: lib/src/screen/dice/dice_screen.dart [hunk 10]
```diff
@@ -22,15 +38,44 @@ class _DiceScreenState extends State<DiceScreen> {
...
+              children: <Widget>[
+                Text(
+                  'Roll a dice:',
+                ),
+                SizedBox(
+                  height: Dimension.diceSizeBoxHeight,
+                ),
+                Container(
+                  child: RollWidget(
+                    futureValue: fetchValue(widget.diceBloc.getRoll()),
```
Comment (luisjavierglb) [inline lib/src/screen/dice/dice_screen.dart]: `snapshot.data`

**Review by MarcosCGlobant — APPROVED** [review]

File: lib/src/app.dart [hunk 11]
```diff
@@ -8,9 +10,32 @@ class MyApp extends StatefulWidget {
...
+
+  @override
+  void dispose() {
+    _blocs.forEach((bloc) {
+      bloc.dispose();
+    });
+    super.dispose();
+  }
+
+  T getDesireBloc<T extends Bloc>() {
```
Comment (luisjavierglb) [inline lib/src/app.dart:33]: typo the word is getDesiredBloc 👍

**Review by luisjavierglb — APPROVED** [review]
LGTM

**Pull request merged**